lncfg/db.go

```diff
@@ -79,54 +84,226 @@ func (db *DB) Init(ctx context.Context, dbPath string) error {
 }
 
 // DatabaseBackends is a two-tuple that holds the set of active database
-// backends for the daemon. The two backends we expose are the local database
-// backend, and the remote backend. The LocalDB attribute will always be
-// populated. However, the remote DB will only be set if a replicated database
-// is active.
+// backends for the daemon. The two backends we expose are the graph database
+// backend, and the channel state backend.
 type DatabaseBackends struct {
-	// LocalDB points to the local non-replicated backend.
-	LocalDB kvdb.Backend
+	// GraphDB points to the database backend that contains the less
+	// critical data that is accessed often, such as the channel graph and
+	// chain height hints.
+	GraphDB kvdb.Backend
 
-	// RemoteDB points to a possibly networked replicated backend. If no
-	// replicated backend is active, then this pointer will be nil.
-	RemoteDB kvdb.Backend
-}
+	// ChanStateDB points to a possibly networked replicated backend that
+	// contains the critical channel state related data.
+	ChanStateDB kvdb.Backend
+
+	// HeightHintDB points to a possibly networked replicated backend that
+	// contains the chain height hint related data.
+	HeightHintDB kvdb.Backend
+
+	// MacaroonDB points to a database backend that stores the macaroon root
+	// keys.
+	MacaroonDB kvdb.Backend
+
+	// DecayedLogDB points to a database backend that stores the decayed log
+	// data.
+	DecayedLogDB kvdb.Backend
+
+	// TowerClientDB points to a database backend that stores the watchtower
+	// client data. This might be nil if the watchtower client is disabled.
+	TowerClientDB kvdb.Backend
+
+	// TowerServerDB points to a database backend that stores the watchtower
+	// server data. This might be nil if the watchtower server is disabled.
+	TowerServerDB kvdb.Backend
+
+	// WalletDB is an option that instructs the wallet loader where to load
+	// the underlying wallet database from.
+	WalletDB btcwallet.LoaderOption
 
-// GetBackends returns a set of kvdb.Backends as set in the DB config.  The
-// local database will ALWAYS be non-nil, while the remote database will only
-// be populated if etcd is specified.
-func (db *DB) GetBackends(ctx context.Context, dbPath string) (
-	*DatabaseBackends, error) {
+	// Replicated indicates whether the database backends are remote, data
+	// replicated instances or local bbolt backed databases.
+	Replicated bool
```

Comment:
I couldn't find any use of this bool anymore besides modifying a log line, so perhaps it can be removed now? Also replicated isn't accurate, because it can be a single instance remote db backend.

Reply:
The decision for the wallet is removed in the second to last commit. But I think the log statement is nice to have. I renamed it to `Remote` though to be more accurate. And I removed it from the root package's `databaseInstances` struct where it's definitely not needed anymore.